app/src/androidTest/java/uk/gov/android/authentication/AppAuthSessionTest.kt

```diff
@@ -1,102 +1,68 @@
-import org.junit.Assert.assertEquals
+import androidx.test.platform.app.InstrumentationRegistry
+import net.openid.appauth.AuthorizationResponse
 import org.junit.Assert.assertThrows
-import org.junit.Ignore
-import org.junit.Test
 import org.mockito.kotlin.any
 import org.mockito.kotlin.mock
 import org.mockito.kotlin.verify
 import org.mockito.kotlin.whenever
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
 
+/* Not checked here:
+* Full route of AppAuthSession::present
+* Full route of AppAuthSession::finalise
+* Because the real AuthorizationService has to have a browser available from the context
+*  */
 class AppAuthSessionTest {
-    private val mockContext: Context = mock()
-    private val mockAuthService: AuthorizationService = mock()
-    private val mockLauncher: ActivityResultLauncher<Intent> = mock()
+    private lateinit var appAuthSession: AppAuthSession
 
-    private val loginSession: LoginSession = AppAuthSession(
-        mockContext
-    )
+    @BeforeTest
+    fun setUp() {
+        val context = InstrumentationRegistry.getInstrumentation().context
+        appAuthSession = AppAuthSession(context)
+    }
 
-    @Ignore("Will be fixed in refactor")
     @Test
-    fun presentLaunchesAuthRequest() {
-        val authUri = Uri.parse("https://example.com/authorize")
-        val redirectUri = Uri.parse("https://example.com/redirect")
-        val tokenUri = Uri.parse("https://example.com/token")
-        val loginSessionConfiguration = LoginSessionConfiguration(
-            authorizeEndpoint = authUri,
-            clientId = "sampleClientId",
-            redirectUri = redirectUri,
-            scopes = listOf(LoginSessionConfiguration.Scope.OPENID),
-            tokenEndpoint = tokenUri,
-            persistentSessionId = "persistentId"
+    fun presentCreatesAndLaunchesAuthorisationIntent() {
+        // Given a mock ActivityResultLauncher
+        val launcher: ActivityResultLauncher<Intent> = mock()
+        whenever(launcher.launch(any())).then {  } // take no action
+        val loginSessionConfig = LoginSessionConfigurationTest.defaultConfig.copy()
+        // When calling present
+        appAuthSession.present(
+            launcher = launcher,
+            configuration = loginSessionConfig
         )
-
-        val expectedIntent = Intent(Intent.ACTION_VIEW)
-        whenever(mockAuthService.getAuthorizationRequestIntent(any()))
-            .thenReturn(expectedIntent)
-
-        loginSession.present(mockLauncher, loginSessionConfiguration)
-
-        verify(mockLauncher).launch(expectedIntent)
+        // Then create the AuthorisationIntent and launch
+        verify(launcher).launch(any())
     }
 
-    @Ignore("Will be fixed in refactor")
-    @Test
-    fun unsuccessfulFinaliseCallbackResponse() {
-        val error = assertThrows(AuthenticationError::class.java) {
-            loginSession.finalise(Intent()) {}
+    @Test(expected = IllegalArgumentException::class)
```

Comment:
This is cool!

Reply:
It's nice if you don't need to check any exception internals, but they've removed it in JUnit5.